include/revng/Model/Type/BaseType.h

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#include "revng/Model/GeneratedType.h"
+
+/// \brief Concept to identify all types that are derived from model::Type
+template<typename T>
+concept IsModelType = std::is_base_of_v<model::Type, T>;
+
+/// \brief Base class of model types used for LLVM-style RTTI
+class model::Type : public model::generated::Type {
+public:
+  // TODO: Constructors cannot be inherited, since the default one is
+  //  manually implemented in order to generate a random ID
+  Type();
+  Type(TypeKind::Values TK);
+  Type(TypeKind::Values Kind, uint64_t ID) : model::generated::Type(Kind, ID) {}
+
+public:
+  static bool classof(const Type *T) { return false; }
+
+  Key key() const { return Key{ Kind, ID }; }
```

Comment:
Is the `Key` type generated in the base class?

Reply:
Yes

Reply:
Who is using this method? If it's used only by generated code, I think we can drop it.

Reply:
Not only by generated code, it's used by `model::Binary::getTypePath` and multiple places in `KeyedObjectContainers.ccp`